pilot/pkg/credentials/kube/secrets.go

```diff
@@ -286,49 +318,3 @@ func extractRoot(scrt *v1.Secret) (cert []byte, err error) {
 	return nil, fmt.Errorf("found secret, but didn't have expected keys %s or %s; found: %s",
 		GenericScrtCaCert, TLSSecretCaCert, found)
 }
-
-func (s *CredentialsController) AddEventHandler(f func(name string, namespace string)) {
```

Comment:
To prevent register handler after informer has been started

Reply:
what is the problem with adding after start? you can add handlers and it will enqueue all objects

Reply:
First start, then register handler, it will miss the notification of some watched objects.

Reply:
In the secret case, seems no influence. Just keep a good manner.

Reply:
nope, IIRC the notification is based on recv from kube-apiserver

Reply:
I will double check it

Reply:
Ah, correct. https://github.com/kubernetes/client-go/blob/8f44946f6cbe967fbe2e2548e76987680a89428e/tools/cache/shared_informer.go#L561-L562)

Reply:
I will revert some unnecessary changes